src/main/java/org/apache/solr/benchmarks/query/DetailedQueryStatsListener.java

```diff
@@ -0,0 +1,99 @@
+package org.apache.solr.benchmarks.query;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.math3.stat.descriptive.SynchronizedDescriptiveStatistics;
+import org.apache.solr.benchmarks.BenchmarksMain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * A Query specific stats listener that accumulates stats for:
+ * <ol>
+ *     <li>Duration of query execution (percentile)</li>
+ *     <li>Document hit count of the query (percentile)</li>
+ *     <li>Error count on the query (long)</li>
+ * </ol>
+ */
+public class DetailedQueryStatsListener implements QueryResponseContentsListener {
+  private final ConcurrentMap<String, SynchronizedDescriptiveStatistics> durationStatsByType = new ConcurrentHashMap<>();
+  private final ConcurrentMap<String, SynchronizedDescriptiveStatistics> docHitCountStatsByType = new ConcurrentHashMap<>();
+  private final ConcurrentMap<String, AtomicLong> errorCountStatsByType = new ConcurrentHashMap<>();
+  private final AtomicBoolean loggedQueryRspError = new AtomicBoolean(false);
+  private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  @Override
+  public void onExecutionComplete(BenchmarksMain.OperationKey key, QueryResponseContents response, long duration) {
+    if (key != null) {
+      SynchronizedDescriptiveStatistics durationStats = durationStatsByType.computeIfAbsent((String) key.getAttributes().get("query"), (typeKey) -> new SynchronizedDescriptiveStatistics());
+      durationStats.addValue(duration / 1_000_000.0);
```

Comment:
it should after successful update, below?

Reply:
I think my thought was that it follows mostly the qtime logic that it handles the duration despite of the result.

I suppose there's a possibility (rather remote tho), that we DO want to benchmark on query failure? 😅 

I understand the concern is that we might overlook failures? Those should be quite obvious when doc hit count is 0 and error count is non zero. It also prints warning to the log